### DISCUSSION TITLE
Madara Hackathon ideas

### DISCUSSION THREAD

**Discussion by AbdelStark** [keep-starknet-strange/madara#374] [Ideas]
# 🚀 Hackathon Ideas for Madara - Let's Build Together! 🌟

Hey there, fellow developers! 👋 We're excited to share with you an amazing opportunity to work with **Madara**, a powerful and versatile open-source StarkNet sequencer built with Rust using the Substrate framework.

Madara's core strength lies in its _community-driven_ approach and flexibility, offering endless possibilities for adding new features and modules. We wholeheartedly embrace the open-source spirit and encourage you to do the same! 🤝

We want to ignite your creativity and inspire you to build innovative and even **crazy** projects using Madara during hackathons. 🎉 Let's collaborate, experiment, and learn together to push the boundaries of what's possible!

Here's what you can do:

1. Share your ideas 💡 for hackathon projects involving Madara.
2. Pick up ideas 💼 from others if you're looking for something to build.
3. Don't hesitate to discuss, ask questions, or seek help from our friendly community. We're all in this together!

Remember, there's no limit to what we can achieve as a collective. Let's build a brighter future for the StarkNet ecosystem and continue to #KeepStarknetStrange! 🚀🌟

_Your contributions make a difference, and we can't wait to see the incredible projects you come up with!_ 🌈

**Comment by AbdelStark** [comment]
# Compatibility with Ethereum addresses
Idea from Louis Guthman

Draft:
```
Create Syscall_deploy_eth_address which takes a ECRecover_eth_sig and deploy at the EthAddress the classHash signed by the user
```

**Comment by AbdelStark** [comment]
# Eigen DA

See this for more details: https://github.com/keep-starknet-strange/madara/discussions/237

**Comment by louisguthmannStarkWare** [comment]
Sovereign Bitcoin Rollups using Ordinals

**Comment by Eikix** [comment]
So someone can:
- build on Madara -> deploy cairo smart contracts during hackathons on a Madara testnet

- upgrade Madara -> add new features to Madara (as plugins? how would someone extend madara? how does it work?) 

- fork Madara and build something cool, c.f. sovereign rollups on bitcoin

**Comment by Eikix** [comment]
Create a Kakarot testnet using Madara and Kakarot, deploy it on the cloud and try out a few transactions!

Links:
- https://github.com/sayajin-labs/kakarot

**Comment by AbdelStark** [comment]
# Support Celestia as Data Availability Layer

**Comment by AbdelStark** [comment]
# Autonomous Smart Contracts

Implement autonomous Starknet smart contracts on Madara.

You can take inspiration from: https://github.com/mikasa-ack
Which is a hackathon project that implemented autonomous smart contracts pallet for Substrate for Ink! based smart contracts. The general process would be the same for Madara, needs some adaptation for Cairo based contracts.
In the Github org of the projects there is a substrate node with the custom pallet, a Game of life frontend, and a Game of life contracts in Ink!.
One could use the same frontend and implement a Game of life in Cairo and implement ASC in Madara.
